### ISSUE TITLE
PerformanceCounter always showing 0 CPU Usage

### ISSUE THREAD

**Issue by Snehal14283** [NLog/NLog#2911]
It display always showing 0 value when using following syntax while in csv file :
`<column name="CPUUsage" layout="${performancecounter:Category=Processor:Counter=% Processor Time:Instance=_Total}"/>`

**Comment by 304NotModified** [comment]
I think it could be access rights. Please check the internal log (and if this still an issue, please fill in the template)

**Comment by Snehal14283** [comment]
No i am having full rights. Please check the below code which is written in C# console application 

`            cpuCounter = new PerformanceCounter("Processor",  "% Processor Time", "_Total");
            cpuCounter.NextValue();
            System.Threading.Thread.Sleep(1000);
            MessageBox.Show("CPU Usage: " + cpuCounter.NextValue() + "%");
`
PerformanceCounter class will give the correct CPU usage on 2nd .NextValue() call. In between we need make thread sleep for at lease 1 sec. On first call of .NextValue() it will always be 0.

As of me, NLog calling .NextValue() and giving the result but on first call it will always be 0. is it true?

**Comment by snakefoot** [comment]
@Snehal14283 You are very correct the normal way of doing performance counters is this:

```
CounterSample currentSample = perfCounter.NextSample();
float perfValue = CounterSample.Calculate(_prevSample, currentSample);
_prevSample = currentSample;
return perfValue;
```

So the `PerformanceCounterLayoutRenderer` doesn't work well with performance-counters that depends on calculation with previous-sample-value (Like CPU usage). Having a Timer ticking every second in the layoutrenderer could be one option, though not sure if it is pretty.

**Comment by 304NotModified** [comment]
What about the `${QPC}` (https://github.com/NLog/NLog/wiki/QPC-Layout-Renderer), is there the same problem? It has a "difference" and "normalize " option

**Comment by snakefoot** [comment]
> What about the ${QPC}

Think that is something different. It is used for high-precision timing measurements. Not related to Processor Usage Performance Counters.

**Comment by Snehal14283** [comment]
I am new to GitHub. Why new thread started #2918 ?  Do its an error in NLog? will i get in next version or do i need to change something in nlog.config?

**Comment by snakefoot** [comment]
@Snehal14283 The new thread is a "Pull Request" which allows one to include source-code-changes.

Sadly enough you have come at a bad time, because right now NLog 4.6 is being prepared, so no official release of NLog in the near feature.

But you are very welcome to test if the pre-release NLog-nuget-package works better (Probably have to clear your nuget-package first to try it out from local nuget-folder):

https://ci.appveyor.com/project/nlog/nlog/build/4.5.0-8367/artifacts

**Comment by 304NotModified** [comment]
fixed by #2918

**Comment by 304NotModified** [comment]
FYI, NLog 4.6 is online which includes a fix for this
